src/rpc/mining.cpp

```diff
@@ -906,16 +911,38 @@ UniValue getblocksubsidy(const UniValue& params, bool fHelp)
 
     UniValue result(UniValue::VOBJ);
     if (canopyActive) {
+        KeyIO keyIO(Params());
         UniValue fundingstreams(UniValue::VARR);
         auto fsinfos = Consensus::GetActiveFundingStreams(nHeight, consensus);
-        for (auto fsinfo : fsinfos) {
+        for (int idx = 0; idx < fsinfos.size(); idx++) {
+            const auto& fsinfo = fsinfos[idx];
             CAmount nStreamAmount = fsinfo.Value(nBlockSubsidy);
             nMinerReward -= nStreamAmount;
 
             UniValue fsobj(UniValue::VOBJ);
             fsobj.pushKV("recipient", fsinfo.recipient);
             fsobj.pushKV("specification", fsinfo.specification);
             fsobj.pushKV("value", ValueFromAmount(nStreamAmount));
+            fsobj.pushKV("valueZat", nStreamAmount);
+
+            auto fs = consensus.vFundingStreams[idx];
+            auto address = fs.get().RecipientAddress(consensus, nHeight);
+
+            CScript* outpoint = boost::get<CScript>(&address);
+            UniValue pubkey(UniValue::VOBJ);
+
+            if (outpoint != nullptr) {
+                // For transparent funding stream addresses
+                ScriptPubKeyToJSON(*outpoint, pubkey, true);
```

Comment:
Why do we want the (JSON representation of the) CScript rather than just the address? The CScript doesn't give any more information, since it's required by ZIP 207 to be of a particular form that only depends on the address.